tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoRepositoryMapping.java

```diff
@@ -125,22 +119,15 @@ protected boolean isValidLocation(String location) {
 
     @Override
     protected File getPrimaryArtifact(File projectRoot) {
-        File[] productFiles = listProducts(projectRoot);
-        if (productFiles.length > 0) {
-            for (File file : productFiles) {
-                if (file.isFile()) {
-                    return file;
-                }
-            }
-        }
-        File category = new File(projectRoot, CATEGORY_XML);
-        if (category.exists()) {
-            return category;
+        Optional<File> repoFile = Optional.empty();
```

Comment:
why assign it here if we overwirte it in the next line? If it is for the exception case, better return the empty optional there. and move the return code into the try block

Reply:
The intention is to check the category.xml even if the product file search failed.
At least before the category.xml was checked if the file search returned an empty error or null (the File.listFiles() ) never throws.
I moved the assignment to the catch clause to make this more clear.
Is that ok?